crates/wit-deps/src/manifest.rs

```diff
@@ -245,7 +245,12 @@ impl Entry {
                     .parent()
                     .with_context(|| format!("`{}` does not have a parent", out.display()))?;
                 lock_deps(ldeps.iter().cloned().map(|id| {
-                    let path = base.join(&id);
+                    // Sanitize dependency name for filesystem compatibility.
+                    // WIT identifiers must be kebab-case (words separated by '-') and cannot contain underscores or other punctuation.
+                    // See: https://component-model.bytecodealliance.org/design/wit.html#identifiers
+                    // We use '-' as the separator to match WIT spec and ensure Windows compatibility.
+                    let sanitized_id = id.replace(':', "-");
```

Comment:
this would cause a clash between `foo:bar-baz` and `foo-bar:baz`, right?

From the linked spec:
> Double hyphens (--) are not allowed.

Can we try using `--` to work around that?

Reply:
again you meant double hyphens?

Reply:
Yes, anything that would be *disallowed* by the WIT spec and *allowed* by most OSes, Windows in particular

Reply:
Right, so I'm thinking if we stored:
- `foo:bar-baz` package -> `foo--bar-baz` directory
- `foo-bar:baz` package -> `foo-bar--baz` directory

Tooling like `wit-bindgen` would be able to read WIT from these directories and rely on WIT package name (`package` directive in WIT).
I don't think any component model tooling actually relies on *directory names* for anything

Reply:
`foo--bar-baz` is not allowed because it contains `--` (continuous double hyphens)

Reply:
> this would cause a clash between foo:bar-baz and foo-bar:baz, right?

Theoretically yes. Practically?